src/test/resources/com/puppycrawl/tools/checkstyle/grammars/javadoc/javadocTags/InputJavadocWithCrAsNewline.txt

```diff
@@ -0,0 +1 @@
+* Test case for detecting CR newlines.
* @author Author Name

```

Comment:
Yes, it looks like you may have to switch all `_tokenStartCharPositionInLine == 0` to `_tokenStartCharPositionInLine == 0 || previousTokenType == NEWLINE`.